NEWS

```diff
@@ -59,8 +59,35 @@ included in the vX.Y.Z section and be denoted as:
    (** also appeared: A.B.C)  -- indicating that this item was previously
                                  included in release version vA.B.C.
 
+4.1.7 -- July, 2024
+-------------------
+
+- Fix bug in Fortran / C interface array calculation.
+- Fix potential initialization in UCX OpenSHMEM support
+- Add OpenSHMEM support for symmetric remote key handling to improve
+  scalability of OpenSHMEM interface.
+- Add MCS Locks as default implementation for OpenSHMEM locks.
+- Fix bug to properly handle ORTE Prefix of "/".
+- Fix bug in I/O interface handling of soft links on Lustre
+  filesystems.
+- Fix memory leak in UCX one-sided implementation.
+- Fix bug in rcache components that could lead to assert on macOS
+  systems.
+- Fix race condition in smcuda Shared Memory transport (used on
+  platforms with CUDA support).
+- Fix mishandling of a NULL completion context in the OFI transport.
+- Tarballs are now built with the tar-pax format.
+- Set MPI_THREAD_SERIALIZED instead of MPI_THREAD_SINGLE as supported
+  threading model when using UCX that supports
+  UCS_THREAD_MODE_SERIALIZED.
+- Improve macOS configure test to only add `-ld_classic` to the
+  Fortran bindings link line if using a version of Xcode that ships a
+  version of the "new linker" that does not support `-commons
+  use_dylibs`.  Xcode 16 betas ship a new linker that supports
+  `-commons use_dylibs`.
+
```

Comment:
@jsquyres I think we probably want to rebase this on top of https://github.com/open-mpi/ompi/pull/12727, and mention in the news that 4.1.6 and older do not build with libfabric 2.0.

Reply:
Is Libfabric 2.0 a thing?  When's it going to be a real thing?  Pre-announcing breakage is.... weird?

Reply:
I think the intention is to clarify that 4.1.6 or older requires libfabric 1.x. In 4.1.7 we fixed a bug so that open mpi can build with newer libfabric source codes.